pkg/tools/grpcutils/url.go

```diff
@@ -102,3 +102,15 @@ func CheckURLFree(u *url.URL) bool {
 	}
 	return err == nil
 }
+
+// CloneURL clones given url
+func CloneURL(u *url.URL) *url.URL {
```

Comment:
Could you please explain, why do we need to clone URL?

Reply:
`grpcutils.ListenAndServe` alters given URL (without actual changes), it results in race conditions when entry restarts:
1. Client connects to `NSMgr.URL`.
2. NSMgr stops.
   * NSMgr restarts and modifies URL.
   * Client starts healing and reconnecting to `NSMgr.URL`.
4. Race condition.